ui/guest/src/classes/ContentController.js

```diff
@@ -290,6 +291,40 @@ export class ContentController {
 
   }
 
+  insertInstance(modelId, fieldId, targetIndex, instance) {
+    const models = this.getCachedModels();
+    const model = models[modelId];
+    // A model with a field that hasn't been initialized with
+    // content may not have that field at all
+    let collection = ModelHelper.value(model, fieldId);
```

Comment:
We need to consider nested here (and everywhere else). Need to use the ModelHelper.extractCollection. There's an example in one of the ops already. Update the other ops that need this too please.

Look for refactoring opportunities between the different operations.